### PR TITLE
refactor(tests): refactoring on tests, remove and add tmpdir

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

File: tests/unit/executors/indexers/test_binary_indexer.py [hunk 1]
```diff
@@ -1,21 +1,10 @@
...
-#         spi.save()
-#
-#     with BinaryPbIndexer.load(spi.save_abspath) as spi:
-#         assert spi.size == num_docs
-#         for j in range(num_docs):
-#             assert spi.query(j) == docs[j]
-
-
-def test_binarypb_in_flow():
+def test_binarypb_in_flow(tmpdir):
```
Comment (maximilianwerk) [inline tests/unit/executors/indexers/test_binary_indexer.py:6]: You could also use the default global fixture, which takes care about setting and deleting the environment variable via
```
@pytest.mark.parametrize('random_workspace_name', ['JINA_TEST_WORKSPACE_BINARY_PB'])
def test_binarypb_in_flow(test_metas):
    ....
```

**Review by maximilianwerk — APPROVED** [review]
Apart from my comment, which you can surely ignore, it LGTM

File: tests/unit/executors/indexers/test_binary_indexer.py (hunk 1 above)
Comment (JoanFM) [reply tests/unit/executors/indexers/test_binary_indexer.py]: Done!

File: tests/unit/test_driver_yaml.py [hunk 2]
```diff
@@ -9,114 +8,110 @@
...
+    assert len(a._drivers) == 4
+    a = BaseExecutor.load_config(resource_filename('jina', '/'.join(('resources', 'executors._pass.yml'))))
+    assert a.name == 'forward'
+    assert len(a._drivers) == 4
+    a = BaseExecutor.load_config(resource_filename('jina', '/'.join(('resources', 'executors._merge.yml'))))
+    assert a.name == 'merge'
+    assert len(a._drivers) == 4
+    a = BaseExecutor.load_config(resource_filename('jina', '/'.join(('resources', 'executors._clear.yml'))))
+    assert a.name == 'clear'
+    assert len(a._drivers) == 4
```
Comment (bwanglzu) [inline tests/unit/test_driver_yaml.py:80]: we can parameterise this one

**Review by bwanglzu — CHANGES_REQUESTED** [review]
the rest looks good!

File: tests/unit/test_driver_yaml.py (hunk 2 above)
Comment (JoanFM) [reply tests/unit/test_driver_yaml.py:80]: Would u mind doing a PR to this branch to see how u would proceed?

Comment (bwanglzu) [reply tests/unit/test_driver_yaml.py:80]:
```python
import pytest

@pytest.mark.parametrize(
    'yaml, name, expected',
    [
        ('executors._route.yml', 'route', 4),
        ('executors._pass.yml',  'forward', 4),
        # the rest ..
    ]
)
def test_resource_executor(yaml, name, expected):
    a = BaseExecutor.load_config(resource_filename('jina', '/'.join(('resources', yaml))))
    assert a.name == name
    assert len(a._drivers) == expected
```

Comment (bwanglzu) [reply tests/unit/test_driver_yaml.py:80]: (I guess this is easier)

Comment (JoanFM) [reply tests/unit/test_driver_yaml.py]: Done! Thanks!

**Review by maximilianwerk — APPROVED** [review]
LGTM

**Pull request merged**